### PR TITLE
chore: add prop-types as a dependency

### PR DESCRIPTION
**Pull request by mobot11**
### Description
We reference `prop-types` in our library but do not list it as a dependency. I _believe_ this works because `annotated-prop-types` lists `prop-types` as a dependency. We should list it as a dependency to avoid future bugs. Thanks @jbadan for pointing this out.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,8 @@
   "dependencies": {
     "classnames": "^2.2.6",
     "fiori-fundamentals": "^1.4.0",
-    "focus-trap-react": "^6.0.0"
+    "focus-trap-react": "^6.0.0",
+    "prop-types": "^15.7.1"
```
Comment (bcullman) [inline package.json]: hrm. this can probably be a devDep.

Comment (mobot11) [reply package.json]: I don't think I agree with that. It needs to be exported in our built code so consumers receive prop-type warnings.

Comment (mobot11) [reply package.json]: https://github.com/facebook/prop-types#how-to-depend-on-this-package

Comment (bcullman) [reply package.json]: im sold. 🚢

**Pull request merged**